draft-specifications/Netlayer-Unix-Domain-Socket.md

```diff
@@ -0,0 +1,123 @@
+              client-challenge>   ; ByteArray of challenge from peer to introduction server
+```
+
+This operation is sent by a netlayer to the introduction server after
+connecting. It contains the OCapN Locator (location) of the peer itself and a
+challenge for the server to sign. The challenge must be some strong randomly
+generated noise that can be used to have the introduction server proove its
+identity to the netlayer.
+
+### uds:server-response
+
+```
+<uds:server-response pubkey                      ; List representing public key in the same format as CapTP
+                     server-challenge            ; ByteArray of challenge from introduction server.
+                     client-challenge-response>  ; uds:signature value
+```
+
+Once the introduction server has read the `uds:register` operation, it MUST
+response with a `uds:server-response`. The contains the following values:
+
+- `pubkey`: The public key of the introduction server which is formatted
+  in the same way as formatted in CapTP. That is following this format:
+  
+  https://github.com/ocapn/ocapn/blob/main/draft-specifications/CapTP%20Specification.md#public-key
+  
+- `server-challenge`: This is like the `client-challenge` a ByteArray containing
+  some strong randomly generated noise which the netlayer will sign to prove its
+  identity.
+  
+- `client-challenge-response`: This is `uds:signature` struct. Please see its
+  definition below.
+
+When the netlayer receives the `uds:server-response` message, it MUST check
+the signature of the client-challenge that it sent against the one provided in
+`client-challenge-response`. The signature MUST be valid for the provided
+server `pubkey`. If the signature is not valid for the `pubkey`, the connection
+MUST be closed. Provided the signature is valid, the netlayer MUST then
+response with a `uds:signature` signing the `server-challenge` provided with its
+designator key.
+
+### uds:signature
+
+```
+<uds:signature payload     ; ByteArray of data being signed
+               signature>  ; List representing signature in the same format as CapTP
+```
+
+This is used both within `uds:server-response` and on it's own by the netlayer
+when signing the `server-challenge`. It includes the following values:
+
+- `payload`: The data being signed.
+- `signature`: A cryptographic signature formatted as a List according to the
+  way CapTP formats signatures. i.e.
+  
+  https://github.com/ocapn/ocapn/blob/main/draft-specifications/CapTP%20Specification.md#signature
+
+When the server receives this in response to its `uds:server-response` message,
+it MUST check that the signature is valid for the key encoded in the OCapN
+peer locator designator. If the signature is valid for the key, the server
+can then finish any setup and provide introductions upon request both to and from
+this netlayer. If the signature is invalid the connection MUST be closed and
+no introductions to the netlayer MUST occur.
```

Comment:
The second "MUST" here seems unusual. Maybe rephrase to something like "If the signature is invalid then introductions cannot occur and the connection MUST be closed."